docs/tutorials/optimize/memory/printf.md

```diff
@@ -1,3 +1,9 @@
-## printf
+# printf and reducing memory
 
-[Optimization document about best practices, guidelines and dos and don'ts for printf]
+The standard library family of `printf` calls (`printf`, `sprintf`, `fprintf` and so on) takes a lot of code space. This is because there are multiple format specifiers, and it is not possible to optimize the code at build time. Even a single `printf` call in your application pulls in the entire standard library.
```

Comment:
```suggestion
The standard library family of `printf` (`printf`, `sprintf`, `fprintf` and so on) calls takes a lot of code space. This is because there are multiple format specifiers, and it is not possible to optimize the code at build time. Even a single `printf` call in your application pulls in the entire standard library.
```